watch.go

```diff
@@ -59,15 +82,29 @@ func (w *Watcher) getEvent() *Event {
 	return w.queue.pop()
 }
 
+func (w *watcher) Watch() WatchChan {
+	return w.watchCh
+}
+
+func (w *watcher) Close() error {
+	w.cancelFunc()
+	return nil
+}
+
 // sendEvent send events to DB's watch
-func (w *Watcher) sendEvent(c chan *Event) {
+func (w *watcher) sendEvent() {
 	for {
-		event := w.getEvent()
-		if event == nil {
-			time.Sleep(100 * time.Millisecond)
-			continue
+		select {
+		case _, ok := <-w.flagCh:
```

Comment:
Why we need the flagCh

Reply:
This is a flag used to record how many events need to be processed, and its size is equal with the number of events saved in event queue. In addition, if the user closes the watch channel, it will notify the goroutine to exit and return.